src/constructs/core/ssm.ts

```diff
@@ -0,0 +1,69 @@
+import type { IStringListParameter, IStringParameter } from "@aws-cdk/aws-ssm";
+import { StringListParameter, StringParameter } from "@aws-cdk/aws-ssm";
+import type { GuStack } from "./stack";
+
+enum SSMParameterType {
+  String,
+  StringList,
+  SecureString,
+}
+
+interface GuSSMParameterProps {
+  version?: number;
+  path?: string; // In practice, some stacks will want to share usage of an account-wide SSM parameter (such as a vpc ID)
+  type: SSMParameterType;
+}
+
+type GuSSMParameterNoTypeProps = Omit<GuSSMParameterProps, "type">;
+type GuSSMSecureStringParameterProps = Omit<GuSSMParameterNoTypeProps, "version">;
+
+function GuSSMParameter(
```

Comment:
Does this mean the [`fromSSM` prop for `GuParameter`](https://github.com/guardian/cdk/blob/cd14aefa11b327bb686a5a379676d2cf4af3da0d/src/constructs/core/parameters.ts#L7) is redundant now?

Reply:
Should these be co-located with the other parameters? Or should the file be called `ssm-parameters` for consistency?

Reply:
Any reason these aren't classes?